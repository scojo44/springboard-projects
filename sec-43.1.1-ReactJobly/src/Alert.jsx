import React from 'react'
import './Alert.css'

export default function Alert({type, messages = []}) {
  return (
    <aside className={`Alert ${type}`}>
      {messages.map(m => <p>{m}</p>)}
    </aside>
  );
}
