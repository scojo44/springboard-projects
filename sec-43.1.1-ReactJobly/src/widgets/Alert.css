.Alert {
  margin: .5rem 2rem;
}
.Alert p {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-radius: .5rem;
  margin: .25rem;
  padding: .5rem;
}
.Alert .success {
  color: white;
  background-color: darkgreen;
}
.Alert .info {
  color: white;
  background-color: darkblue;
}
.Alert .error {
  color: white;
  background-color: maroon;
}
.Alert button {
  background-color: rgba(0,0,0,10%); 
  border: none;
}
