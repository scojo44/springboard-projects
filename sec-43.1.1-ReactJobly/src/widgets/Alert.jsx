import React from 'react'
import './Alert.css'

export default function Alert({alerts, dismiss}) {
  return (
    <aside className="Alert">
      {alerts.map((alert, i) =>
        <p className={alert.category} key={i}>
          <span>{alert.message}</span>
          <button type="button" onClick={() => dismiss(alert)}>X</button>
        </p>
      )}
    </aside>
  );
}
