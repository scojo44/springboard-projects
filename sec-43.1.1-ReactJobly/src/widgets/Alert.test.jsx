import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import Alert from './Alert'

describe('Alert Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <Alert type="success" messages={['test alert']} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <Alert type="success" messages={['test alert']} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
