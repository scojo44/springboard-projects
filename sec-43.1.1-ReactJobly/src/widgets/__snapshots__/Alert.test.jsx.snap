// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <aside
    class="Alert"
  />
</DocumentFragment>
`;
